src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -47,7 +47,8 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                     return true;
                 }
             }
-            else if (reader.TryGetInt64(out long longValue))
+
+            if (reader.TryGetInt64(out long longValue))
```

Comment:
Can you add a test with a value that is in the `UInt64` range but not in the `Int64` range so something like `9223372036854775808` and a negative value that doesn't fit in `Int64` as well: `-9223372036854775809`.

Similarly, a negative int larger than UInt64, in magnitude: `-18446744073709551617`
https://github.com/dotnet/corefx/blob/d24f68c8ebdcf52df954f60f3f1d3654492a49b7/src/System.Text.Json/tests/Serialization/EnumTests.cs#L41

Reply:
If it's in UInt64 but not Int64 range it's ok will be handled like UInt64...I think you're right for negative out of Int64, I'm OOF at the moment tomorrow I'll do tests.
I think that here we need to chose how handle...let overflow/underflow and throw if not convertible...avoid overflow/underflow, check min/max bound(with fallbacks for instance for -1) and throw.

Reply:
> If it's in UInt64 but not Int64 range it's ok will be handled like UInt64

Not if the enum is backed by an Int64.

Reply:
I think we should throw an exception for values outside the bounds of the underlying enum (as I described here: https://github.com/dotnet/corefx/issues/38363#issuecomment-500164282). We can change from throwing an exception to allowing (and overflowing) in the future if there is a clear user scenario. We can't go the other way, so it is safest to start with a stricter policy.

Reply:
> Not if the enum is backed by an Int64.

Yes in that case, sorry misread

Reply:
I'll add new tests with all out of bounds int,uint,ulong positive negative with expected assertions.

Reply:
> I think that here we need to chose how handle...let overflow/underflow and throw if not convertible...avoid overflow/underflow, check min/max bound(with fallbacks for instance for -1) and throw.

> I think we should throw an exception for values outside the bounds of the underlying enum 

To clarify, we should throw for overflow values that are outside the range of the backing numeric type of the enum (int/uint/etc.). We should continue to allow "in-range" values even if those values do not exist within the enum itself. So for deserializing a number `5` to `enum : int {A, B,C}` should work (even though the enum contains only 3 values), but number `int.MaxValue + 1` should throw.